src/pages/[platform]/build-a-backend/data/data-modeling/secondary-index/index.mdx

```diff
@@ -63,6 +65,44 @@ const { data, errors } =
   // highlight-end
 ```
 
+</InlineFilter>
+
+<InlineFilter filters={["swift"]}>
+
+The example client query below creates a custom GraphQL request that allows you to query for "Customer" records based on their `accountRepresentativeId`:
+
+```swift
+struct PaginatedList<ModelType: Model>: Decodable {
+    let items: [ModelType]
+    let nextToken: String?
+}
+let operationName = "listCustomer8ByAccountRepresentativeId"
+let document = """
+query ListCustomer8ByAccountRepresentativeId {
+  \(operationName)(accountRepresentativeId: "\(accountRepresentativeId)") {
+    items {
+      createdAt
+      accountRepresentativeId
+      id
+      name
+      phoneNumber
+      updatedAt
+    }
+    nextToken
+  }
+}
+"""
+
+let request = GraphQLRequest<PaginatedList<Customer>>(
+    document: document,
+    responseType: PaginatedList<Customer>.self,
+    decodePath: operationName)
+
+let queriedCustomers = try await Amplify.API.query(
+    request: request).get()
+```
+
+</InlineFilter>
```

Comment:
<img width="1017" alt="image" src="https://github.com/aws-amplify/docs/assets/1365977/23c76ef1-4a5b-4833-99b8-874ba01344d7">

- A custom document is needed here since the Swift client does not contain any knowledge of the new list query operation. 
- PaginatedList shows how to decode from the graphQL response shape containing `items` and `nextToken`.